marshmallow/fields.py

```diff
@@ -639,6 +639,8 @@ class Number(Field):
     """Base class for number fields.
 
     :param bool as_string: If True, format the serialized value as a string.
+    :param bool prohibit_boolean: If True, attempting to serialize a boolean
```

Comment:
I don't think we need to add this as a parameter. `Number` should only accept numbers as input; booleans should be considered invalid.